### ISSUE TITLE
Cannot determine completion type for current selection, for multi cursors

### ISSUE THREAD

**Issue by evandrocoan** [SublimeText/LaTeXTools#1161]
I created this command I ran using the [Chain Of Command](https://github.com/jisaacks/ChainOfCommand) package on my user folder, based on:

1. https://github.com/SublimeText/LaTeXTools/issues/1156 Ability to insert project already existent images command by a quick panel

(User/Commands/LaTeX.sublime-commands)
```javascript
[
	{
		"caption": "LaTeXTools: Insert Image",
		"command": "chain",
		"args": {
		    "commands": [
		        ["insert_snippet", {"contents": "\\begin{figure}\n\\centering\n\\includegraphics[scale=1.0]{$TM_SELECTED_TEXT$1}\n\\caption{$1}\n\\label{fig:$1}\n\\end{figure}"}],
		        ["latex_fill_all"]
		    ]
		},
	},
]
```
This creates a selection like this:

![image](https://user-images.githubusercontent.com/5332158/27099832-945a6782-5052-11e7-914e-f0fae9bdfe84.png)

However when there are multiple cursors, the Sublime Text console receives a:
```
Cannot determine completion type for current selection
```
When I run the command `LaTeXTools: Insert Image`.

___

Related issues:

1. https://github.com/SublimeText/LaTeXTools/issues/815 Cite-autocompletion no longer working

**Comment by r-stein** [comment]
Yeah you need to use `$1` only inside includegraphics and $2 inside the other ones.

**Comment by evandrocoan** [comment]
But I want to it to autocomplete the image name/path on all these fields. Currently I can do this with the smart paste feature, setting it to `<<image_name>>` 3 times at `User\LaTeXTools (Advanced).sublime-settings`:
```javascript
{
    "smart_paste_image_snippet": [
        "\\begin{figure}[H]",
        "\\centering",
        "\\includegraphics[scale=1.0$2]{<<image_name>>}",
        "\\caption{<<image_name>>$TM_SELECTED_TEXT$1}",
        "\\label{fig:<<image_name>>${1/([0-9a-zA-Z_])|([^0-9a-zA-Z_]+)/\\L\\1(?2:_:)/g}}",
        "\\end{figure}",
    ],
}
```

**Comment by ig0774** [comment]
You could try changing the `"latex_fill_all"` to something like `"latex_fill_all", {"completion_type": "input", "force": false}`, which may work (though I can't make any guarantees... the FillAll bit wasn't really designed to accommodate this sort of thing).

**Comment by evandrocoan** [comment]
I tried to use your suggesting setting force to `false`, but not success. Seems you still need to specify to ignore multiple selections, otherwise it just "cannot guess" the completion. However as I am using `force` true, I think it should "force" for the `completion_type` I am setting. However I could not understand the fill all implementation fully as it is pretty big and would resort to a higher amount of time to study it.

**Comment by deathaxe** [comment]
Image file is auto-completed for all 3 fields with LaTeXTools 4.5.8

Closing as fixed.
